tests/tests_geomstats/test_spd_matrices.py

```diff
@@ -19,559 +21,548 @@
     SPDMetricEuclidean,
     SPDMetricLogEuclidean,
 )
+from tests.conftest import Parametrizer, TestCase, TestData
 
-SQRT_2 = math.sqrt(2)
+SQRT_2 = math.sqrt(2.0)
+LN_2 = math.log(2.0)
+EXP_1 = math.exp(1.0)
+EXP_2 = math.exp(2.0)
+SINH_1 = math.sinh(1.0)
 
 
-class TestSPDMatrices(geomstats.tests.TestCase):
+class TestSPDMatrices(TestCase, metaclass=Parametrizer):
     """Test of SPDMatrices methods."""
 
-    def setup_method(self):
-        """Set up the test."""
-        warnings.simplefilter("ignore", category=ImportWarning)
-
-        gs.random.seed(1234)
-
-        self.n = 3
-        self.space = SPDMatrices(n=self.n)
-        self.metric_affine = SPDMetricAffine(n=self.n)
-        self.metric_bureswasserstein = SPDMetricBuresWasserstein(n=self.n)
-        self.metric_euclidean = SPDMetricEuclidean(n=self.n)
-        self.metric_logeuclidean = SPDMetricLogEuclidean(n=self.n)
-        self.n_samples = 4
-
-    def test_belongs(self):
-        """Test of belongs method."""
-        mats = gs.array([[3.0, -1.0], [-1.0, 3.0]])
-        result = SPDMatrices(2).belongs(mats)
-        expected = True
-        self.assertAllClose(result, expected)
-
-        mats = gs.array([[-1.0, -1.0], [-1.0, 3.0]])
-        result = SPDMatrices(2).belongs(mats)
-        expected = False
-        self.assertAllClose(result, expected)
-
-        mats = gs.eye(3)
-        result = SPDMatrices(2).belongs(mats)
-        expected = False
-        self.assertAllClose(result, expected)
-
-    def test_belongs_vectorization(self):
-        """Test of belongs method."""
-        mats = gs.array(
-            [[[1.0, 0], [0, 1.0]], [[1.0, 2.0], [2.0, 1.0]], [[1.0, 0.0], [1.0, 1.0]]]
+    class TestDataSPDMatrices(TestData):
+        def belongs_data(self):
+            smoke_data = [
+                dict(n=2, mat=[[3.0, -1.0], [-1.0, 3.0]], expected=True),
```

Comment:
Just thought about it: could you also adapt `test_symmetric_matrices` so that we always use `expected` ? I see some tests that do not respect this convention here:
https://github.com/geomstats/geomstats/blob/baf78ca203355cae2ea9403e8180fc13deed55ea/tests/tests_geomstats/test_symmetric_matrices.py#L37

Reply:
sure